packages/schematics/angular/migrations/update-8/index.ts

```diff
@@ -0,0 +1,128 @@
+  'use-host-property-decorator': 'no-host-metadata-property',
+  'use-input-property-decorator': 'no-inputs-metadata-property',
+  'use-output-property-decorator': 'no-outputs-metadata-property',
+  'no-queries-parameter': 'no-queries-metadata-property',
+  'pipe-impure': 'no-pipe-impure',
+  'use-view-encapsulation': 'use-component-view-encapsulation',
+  i18n: 'template-i18n',
+  'banana-in-box': 'template-banana-in-box',
+  'no-template-call-expression': 'template-no-call-expression',
+  'templates-no-negated-async': 'template-no-negated-async',
+  'trackBy-function': 'template-use-track-by-function',
+  'no-attribute-parameter-decorator': 'no-attribute-decorator',
+  'max-inline-declarations': 'component-max-inline-declarations',
+};
+
+function updateTsLintConfig(): Rule {
+  return (host: Tree) => {
+    const tsLintPath = '/tslint.json';
+    const buffer = host.read(tsLintPath);
+    if (!buffer) {
+      return host;
+    }
+    const tsCfgAst = parseJsonAst(buffer.toString(), JsonParseMode.Loose);
+
+    if (tsCfgAst.kind != 'object') {
+      return host;
+    }
+
+    const rulesNode = findPropertyInAstObject(tsCfgAst, 'rules');
+    if (!rulesNode || rulesNode.kind != 'object') {
+      return host;
+    }
+
+    const recorder = host.beginUpdate(tsLintPath);
+
+    rulesNode.properties.forEach(prop => {
+      const mapping = ruleMapping[prop.key.value];
+      if (mapping) {
+        recorder.remove(prop.key.start.offset + 1, prop.key.value.length);
+        recorder.insertLeft(prop.key.start.offset + 1, mapping);
+      }
+    });
+
+    host.commitUpdate(recorder);
+
+    return host;
+  };
+}
+
+function updatePackageJson(config: CliConfig) {
+  return (host: Tree, context: SchematicContext) => {
+    const dependency: NodeDependency = {
+      type: NodeDependencyType.Dev,
+      name: 'codelyzer',
+      version: '^5.0.0',
+      overwrite: true,
+    };
+
+    addPackageJsonDependency(host, dependency);
+
+    context.addTask(new NodePackageInstallTask({
+      packageManager: config.packageManager === 'default' ? undefined : config.packageManager,
+    }));
+
+    return host;
+  };
+}
+
+function getConfigPath(tree: Tree): Path {
+  const possiblePath = normalize('angular-cli.json');
+  if (tree.exists(possiblePath)) {
+    return possiblePath;
+  }
+
+  throw new SchematicsException('Could not find configuration file');
+}
+
+export default function(): Rule {
+  return (host: Tree) => {
+    const configPath = getConfigPath(host);
+    const configBuffer = host.read(normalize(configPath));
+    if (configBuffer == null) {
+      throw new SchematicsException(`Could not find configuration file (${configPath})`);
+    }
+
+    const config = parseJson(configBuffer.toString(), JsonParseMode.Loose);
+    if (typeof config != 'object' || Array.isArray(config) || config === null) {
+      throw new SchematicsException('Invalid angular-cli.json configuration; expected an object.');
```

Comment:
If passing the package manager is still needed, the config file here should be `angular.json` and not `angular-cli.json`